fetch-examples/src/test/scala/Http4sExample.scala

```diff
@@ -149,7 +149,7 @@ class Http4sExample extends AnyWordSpec with Matchers {
     def fetch[F[_]: Async]: Fetch[F, List[(User, List[Post])]] =
       for {
         users          <- fetchManyUsers(List(1, 2))
-        usersWithPosts <- users.traverse(fetchPosts[F])
+        usersWithPosts <- Fetch.batchAll(users.map(fetchPosts[F]): _*)
```

Comment:
What do you think about providing some syntax sugar on `Seq[Fetch[F[_], A]]` allowing users to write this as `users.map(fetchPosts[F]).batchAll`?

Reply:
Added `.batchAll` and `.batchAllWith` to mimic `.sequence` and `.traverse` respectively for batches